sensors/weather/sensorhub-driver-kestrel/src/main/java/org/sensorhub/impl/sensor/kestrel/Kestrel.java

```diff
@@ -0,0 +1,378 @@
+import org.sensorhub.api.comm.ble.IGattField;
+import org.sensorhub.api.comm.ble.IGattService;
+import org.sensorhub.api.common.SensorHubException;
+import org.sensorhub.impl.sensor.AbstractSensorModule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.ref.WeakReference;
+import java.nio.ByteBuffer;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.UUID;
+
+
+/**
+ *
+ * @author Kalyn Stricklin
+ * @since Dec 1, 2025
+ */
+public class Kestrel extends AbstractSensorModule<KestrelConfig> {
+    static final Logger logger = LoggerFactory.getLogger(Kestrel.class.getSimpleName());
+    EnvironmentalOutput environmentalOutput;
+    private static final UUID ENVIRONMENTAL_SERVICE = UUID.fromString("03290000-eab4-dea1-b24e-44ec023874db");
+    private static final UUID SENSOR_MEASUREMENTS_CHAR = UUID.fromString("03290310-eab4-dea1-b24e-44ec023874db");
+    private static final UUID DERIVED_MEASUREMENTS_1_CHAR = UUID.fromString("03290320-eab4-dea1-b24e-44ec023874db");
+    private static final UUID DERIVED_MEASUREMENTS_2_CHAR = UUID.fromString("03290330-eab4-dea1-b24e-44ec023874db");
+    private static final UUID DERIVED_MEASUREMENTS_3_CHAR = UUID.fromString("03290340-eab4-dea1-b24e-44ec023874db");
+    private static final UUID DERIVED_MEASUREMENTS_4_CHAR = UUID.fromString("03290350-eab4-dea1-b24e-44ec023874db");
+    private static final UUID CLIENT_CONFIG = UUID.fromString("00002902-0000-1000-8000-00805f9b34fb");
+
+    private boolean btConnected;
+    private IGattCharacteristic sensorMeasurements;
+    private IGattCharacteristic derivedMeasurementsOne;
+    private IGattCharacteristic derivedMeasurementsTwo;
+    private IGattCharacteristic derivedMeasurementsThree;
+    private IGattCharacteristic derivedMeasurementsFour;
+
+    IGattClient gattClient;
+
+    WeakReference<IBleNetwork<?>> bleNetRef;
+    KestrelEnvData env = new KestrelEnvData();
+
+
+    @Override
+    protected void doInit() throws SensorHubException {
+        super.doInit();
+
+        // generate IDs
+        generateUniqueID("urn:osh:kestrel:", config.serialNumber);
+        generateXmlID("KESTREL_WEATHER", config.serialNumber);
+
+        // create output interfaces
+        addOutputs();
+    }
+
+    private void addOutputs() {
+        environmentalOutput = new EnvironmentalOutput(this);
+        environmentalOutput.doInit();
+        addOutput(environmentalOutput, false);
+    }
+
+    @Override
+    public void doStart() throws SensorHubException {
+
+        if (bleNetRef == null) {
+            var moduleRegistry = getParentHub().getModuleRegistry();
+            bleNetRef = moduleRegistry.getModuleRef(config.networkID);
+            if (bleNetRef != null) {
+                System.out.println("connecting to gatt");
+                bleNetRef.get().connectGatt(config.deviceAddress, gattCallback);
+            }
+
+        }
+    }
+
+    Queue<IGattDescriptor> descriptorWriteQueue = new LinkedList<>();
```

Comment:
If this linked list is for asynchronous processing use a thread safe alternative like LinkedBlockingQueue